src/payment/strategies/apple-pay/apple-pay-payment-strategy.ts

```diff
@@ -0,0 +1,166 @@
+import { RequestSender } from '@bigcommerce/request-sender';
+
+import { Cart } from '../../../cart';​
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError, PaymentMethodCancelledError } from '../../errors';
+import Payment from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethod from '../../payment-method';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import ApplePaySessionFactory from './apple-pay-session-factory';
+import { assertApplePayWindow } from './is-apple-pay-window';
+
+export const validationEndpoint = 'https://bigpay.service.bcdev/api/public/v1/payments/applepay/validate_merchant';
+const appleValidationUrl = 'https://apple-pay-gateway-cert.apple.com/paymentservices/startSession';
+
+interface ApplePayPromise {
+    resolve(value: InternalCheckoutSelectors | PromiseLike<InternalCheckoutSelectors>): void;
+    reject(reason?: Error): void;
+}
+​
+export default class ApplePayPaymentStrategy implements PaymentStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _requestSender: RequestSender,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _sessionFactory: ApplePaySessionFactory
+    ) { }
+​
+    initialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+​
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment } = payload;
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        const request = this._getBaseRequest(cart);
+        const applePaySession = this._sessionFactory.create(request);
+​
+        const { methodId } = payment;
+        const {
+            paymentMethods: { getPaymentMethodOrThrow },
+        } = await this._store.dispatch(
+            this._paymentMethodActionCreator.loadPaymentMethod(methodId, options)
+        );
+
+        const paymentMethod = getPaymentMethodOrThrow(methodId);
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder({}, options));
+​
+        assertApplePayWindow(window);
+​
+        applePaySession.begin();
+​
+        // Applepay will handle the rest of the flow so return a promise that doesn't really resolve
```

Comment:
Is this comment still relevant? I can see that `_handleApplePayEvents` does call `resolve` in `onpaymentauthorized` handler?

Reply:
Removing it now.